src/game/engine/unit.cpp

```diff
@@ -288,14 +422,35 @@ DirType UnitClass::Desired_Load_Dir(ObjectClass *object, cell_t &cellnum) const
 #endif
 }
 
+/**
+ *
+ *
+ */
 DirType UnitClass::Fire_Direction() const
 {
-#ifdef GAME_DLL
-    DEFINE_CALL(func, 0x005803C4, DirType, const UnitClass *);
-    return func(this);
-#else
-    return DIR_NONE;
-#endif
+    if (Class_Of().Is_Turret_Equipped()) {
+        // Special logics for the V2 facing.
+        if (What_Type() == UNIT_V2RL) {
+            // TODO figure out what exactly is going on here and document it.
```

Comment:
Its working out which is the minimum distance needed to turn by trying both directions.